relay-wstring/src/lib.rs

```diff
@@ -0,0 +1,512 @@
+//! A UTF-16 little-endian string type.
+//!
+//! The main type in this crate is [WStr] which is a type similar to [str] but with the
+//! underlying data encoded as UTF-16 with little-endian byte order.
+//!
+//! # Examples
+//!
+//! ```
+//! use relay_wstring::WStr;
+//!
+//! let b = b"h\x00e\x00l\x00l\x00o\x00";
+//! let s: &WStr = WStr::from_utf16le(b).unwrap();
+//!
+//! let chars: Vec<char> = s.chars().collect();
+//! assert_eq!(chars, vec!['h', 'e', 'l', 'l', 'o']);
+//!
+//! assert_eq!(s.to_utf8(), "hello");
+//! ```
+
+#![deny(missing_docs, missing_debug_implementations)]
+
+use std::error::Error;
+use std::fmt;
+use std::iter::FusedIterator;
+use std::slice::ChunksExact;
+
+mod slicing;
+
+#[doc(inline)]
+pub use crate::slicing::SliceIndex;
+
+/// Error for invalid UTF-16 encoded bytes.
+#[derive(Debug, Copy, Clone)]
+pub struct Utf16Error {}
+
+impl fmt::Display for Utf16Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "Invalid UTF-16LE data in byte slice")
+    }
+}
+
+impl Error for Utf16Error {}
+
+impl Utf16Error {
+    /// Create a new [Utf16Error].
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+impl Default for Utf16Error {
```

Comment:
Does it make sense to implement `Default`? If we add any field to the error, this default probably has to go.